mvlearn/semi_supervised/ctregression.py

```diff
@@ -0,0 +1,580 @@
+# python version 3.7.6
+
+import pandas as pd
+import numpy as np
+from sklearn.neighbors import KNeighborsRegressor
+import random
+from scipy.spatial.distance import minkowski
+from ..utils.utils import check_Xs, check_Xs_y_nan_allowed
+from .base import BaseCoTrainEstimator
+
+# Implements Multi view Co training Regression for 2 view data
+
+
+class CTRegressor(BaseCoTrainEstimator):
+    r"""
+    This class implements the co-training regression for supervised
+    and semisupervised learning with the framework as described in [#1CTR].
+    The best use case is when nthe 2 views of input data are sufficiently
+    distinct and independent as detailed in [#1CTR]. However this can also
+    be successfull when na single matrix of input data is given as
+    both views and two estimators are choosen
+    which are quite different.[#2CTR]
+
+    In the semi-supervised case, performance can vary greatly, so using
+    a separate validation set or cross validation procedure is
+    recommended to ensure the regression model has fit well.
+
+    Parameters
+    ----------
+    estimator1: sklearn object, (currently only supports K Nearest Neighbour)
+        The regressor object which will be trained on view1 of the data.
+
+    estimator2: sklearn object, (currently only support K Nearest Neighbour)
+        The regressir object which will be trained on view2 of the data.
+
+    neighbors_size: int, optional (default = 5)
+        The number of neighbours to be considered for determining the mean
+        squared error.
+
+    supsample_size: int, optional (default = 50)
```

Comment:
to keep this consistent with CTClassifier, this should probably be called unlabeled_pool_size